Sources/xcodeproj/Objects/Project/PBXProjEncoder.swift

```diff
@@ -16,11 +56,20 @@ final class PBXProjEncoder {
     var indent: UInt = 0
     var output: String = ""
     var multiline: Bool = true
-
+    
+    public var fileListSortOrder: FileListSortOrder = .byUUID
+    public var navigatorGroupSortOrder: ProjectNavigatorSortOrder = .unsorted
+    public var buildPhaseFileSortOrder: BuildPhaseFileSortOrder = .unsorted
+    
     // swiftlint:disable function_body_length
     func encode(proj: PBXProj) throws -> String {
+        
         try referenceGenerator.generateReferences(proj: proj)
         guard let rootObject = proj.rootObjectReference else { throw PBXProjEncoderError.emptyProjectReference }
+        
+        // Sort sub objects in advance.
+        proj.forEach(self.sortFiles)
```

Comment:
Performance-wise, you are iterating through all the objects in the project, even if you don't plan to sort them. What I suggest is getting the objects that you plan to sort and execute your sorting logic on them. Note that [there are getters](https://github.com/tuist/xcodeproj/blob/master/Sources/xcodeproj/Objects/Project/PBXProj.swift#L88) for all different types of objects that a project has.

Reply:
Yep. Good idea. I think originally I thought I'd have to deal with a much larger range of types. Refactoring.